src/main/java/io/netty/incubator/codec/http3/Http3ConnectionHandler.java

```diff
@@ -18,21 +18,24 @@
 import io.netty.channel.ChannelHandler;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.incubator.codec.http3.Http3FrameCodec.Http3FrameCodecFactory;
 import io.netty.incubator.codec.quic.QuicChannel;
 import io.netty.incubator.codec.quic.QuicStreamChannel;
 import io.netty.incubator.codec.quic.QuicStreamType;
 
-import java.util.function.Function;
 import java.util.function.LongFunction;
 
 /**
  * Handler that handles <a href="https://tools.ietf.org/html/draft-ietf-quic-http-32">HTTP3</a> connections.
  */
 public abstract class Http3ConnectionHandler extends ChannelInboundHandlerAdapter {
-    private final Function<Http3FrameTypeValidator, Http3FrameCodec> codecFactory;
+    private final Http3FrameCodecFactory codecFactory;
```

Comment:
@normanmaurer I added this factory for passing the stream type but didn't need it eventually. However, I thought I will keep the factory instead of the `Function` so that it is easier to modify the args later. These are all package private interfaces and fields so I think it is Ok. WDYT?